### ISSUE TITLE
noreply_wait waits for non-noreply queries

### ISSUE THREAD

**Issue by AtnNn** [rethinkdb/rethinkdb#3812]
I would expect it not to.

Reproducible with this script:

```
r = require 'rethinkdb'

r.connect port:28019, (e, c) ->
  console.log new Date(), "WAITING"
  c.noreplyWait ->
    console.log new Date(), "DONE"
    r.range(100000000).sum().run c, (e, r) -> console.log new Date(), "RESULT", r
    console.log new Date(), "WAITING"
    c.noreplyWait -> console.log new Date(), "DONE"
```

```
$ coffee test.coffee
16:29:48 'WAITING'
16:29:48 'DONE'
16:29:48 'WAITING'
16:30:01 'RESULT' 4999999950000000
16:30:01 'DONE'
```

**Comment by danielmewes** [comment]
I don't think this is critical, but would be nice to fix.

Assigning @Tryneus

**Comment by Tryneus** [comment]
The fix for this is up in review 2720.  It turns out this is a problem we need to deal with because closing a connection with open changefeed requests using `conn.close(noreply_wait=True)` will hang indefinitely.

**Comment by Tryneus** [comment]
The fix has been merged to `next` in commit fe575b6 and cherry-picked into `v2.0.x` in commit d786b8d.  Will be in release 2.0.
